File: solid principle/SolidPrinciple.java

class Rectangle{
    public int height;
    public int width;

    public void setWidth(int width) {
        this.width = width;
    }

    public void setheight(int height) {
        this.height = height;
    }

    public int getArea() {
        return this.height * this.width;
    }

}

class Square extends Rectangle {

    @Override
    public void setWidth(int width) {
        super.width = width;
        super.height = width;
    }
    @Override
    public void setheight(int height) {
        super.height = height;
        super.width = height;
    }

}
public class SolidPrinciple {

    public static void main(String[] args) {

        //wrong implementation
        Rectangle r= new Square();
        r.setheight(3);
        r.setWidth(3);
        System.out.println(r.getArea());

       //wrong rectangle is not square 
       //subclass should behave in a way that does not break the expectations set by its superclass.
        //Rectangles r= new Squares();
        //r.setSide(4);
        //System.out.println(r.getArea());

    }

    //correct implementation
    interface Shape{
        int getArea();
    }

    class Rectangles implements Shape{

        private int height;
        private int width;

        
        @Override
        public int getArea() {
            // TODO Auto-generated method stub
           return height*width;
        }

        public int getHeight() {
            return height;
        }

        public void setHeight(int height) {
            this.height = height;
        }

        public int getWidth() {
            return width;
        }

        public void setWidth(int width) {
            this.width = width;
        }

    }

    class Squares implements Shape{

        private int side;
       
        @Override
        public int getArea() {
            // TODO Auto-generated method stub
           return side*side;
        }

        public int getSide() {
            return side;
        }

        public void setSide(int side) {
            this.side = side;
        }

    }
}

File: solid principle/SolidPrinciple.py

class Rectangle_:
    
    def __init__(self, h:int, w:int):
        self.w=w
        self.h=h

    def getArea(self)->int:
        return self.h * self.w
    



class Square_(Rectangle_):
    
     def __init__(self, s:int):
       super().__init__(s,s)


     def getArea(self)->int : 
        return self.h*self.w
    
 
if __name__ == "__main__":    
    r=Rectangle_(5,8).getArea()
    print(r)    
    s=Square_(5).getArea()
    print(s)    
    


from abc import ABC, abstractmethod

# Define an abstract base class for Shape
class Shape(ABC):

    @abstractmethod
    def get_area(self) -> int:
        pass  # Enforce implementation in subclasses


# Rectangle class inheriting from Shape
class Rectangle(Shape):
    def __init__(self, height: int, width: int):
        self.height = height
        self.width = width

    def get_area(self) -> int:
        return self.height * self.width


# Square class inheriting from Shape
class Square(Shape):
    def __init__(self, side: int):
        self.side = side

    def get_area(self) -> int:
        return self.side * self.side


# Example usage
rect = Rectangle(5, 10)
print("Rectangle Area:", rect.get_area())

sq = Square(4)
print("Square Area:", sq.get_area())

